examples/parametertree.py

```diff
@@ -145,11 +154,11 @@ def valueChanging(param, value):
         ch2.sigValueChanging.connect(valueChanging)
         
 
-
 def save():
     global state
     state = p.saveState()
-    
+QtCore.QTimer.singleShot(0, save)
```

Comment:
We had really weird issues involving `QTimer.singleShot()` (see #1605 )

Would suggest creating a timer, and explicitly calling `setSingleShot`  You can see an example here: https://github.com/pyqtgraph/pyqtgraph/pull/1605/commits/85e894dd73d21b479bcc0d98431e17ed88d583d3

Reply:
Would you object to a `safeSingleShot` function in `functions.py` that spawns a `QTimer` this way, if the paradigm needs to be reused everywhere?

Reply:
Why is the delayed call to ```save()``` needed? You have the following right before ```pg.exec()```
```python
## test save/restore
state = p.saveState()
p.restoreState(state)
```

Reply:
IIRC, there was an issue where the restored state didn't match the actual state before saving, so I delayed the initial save for things like the gradient editor's size to finalize. But testing it now, the `singleShot` seems unnecessary.